[stable/24.03] COS integration

mkalcok opened the pull request:
This is proposal preview that builds on top of #9 and introduces COS integration to branch `24.03/stable`. Once the prerequisite PR #9 is merged, this change will be rebased.

mkalcok committed:
tox: Use ZAZA_FEATURE_BUG472 in tests.

This environment variable allows functional tests to correctly
get unit's IP addres.

(cherry picked from commit 8b736860d53f00213039f306ba4bfdf937bc64b2)
Signed-off-by: Martin Kalcok <martin.kalcok@canonical.com>

mkalcok committed:
build.lock: Update layer-ovn to include COS.

Newer version of layer-ovn is needed for COS integration.

Signed-off-by: Martin Kalcok <martin.kalcok@canonical.com>

mkalcok committed:
functest: COS integration.

This commit  enables COS integration functional tests
for branch 24.03/stable. It includes:
  * bundles and overlays for cos-lite deployment
  * jammy-caracal test bundle updated to test integration with COS
  * Github CI actions updated to test integration with COS

func-test-pr: https://github.com/openstack-charmers/zaza-openstack-tests/pull/1324
Signed-off-by: Martin Kalcok <martin.kalcok@canonical.com>

mkalcok committed:
deps: Upgrade locked dependencies to juju3

COS functional tests will require juju3.

Signed-off-by: Martin Kalcok <martin.kalcok@canonical.com>

fnordahl reviewed (dismissed):
LGTM, thanks!

Note that openstack-charmers/zaza-openstack-tests#1324 is a dependency and needs to be merged prior to merging this one.

mkalcok commented:
Ah, a subtle issue was introduced in https://github.com/canonical/charm-ovn-chassis/pull/9/commits/e19795ad8deafcddcd5fa08c7c1b400c1051e1af. The model `jammy-caracal` was tagged/named `jammy_caracal` in `tests.yaml`, but then no configure/test steps were explicitly defined for it. This caused zaza to only deploy the mode but not run any tests.
I'll post the fix once I verify it locally.

mkalcok committed:
test: Explicitly specify tests for jammy-caracal

Due to the way zaza works, when a bundle is named (e.g
'jammy_caracal: jammy-caracal'), to apply bundle-specific options,
all configure and test steps need to be explicitly defined as well.
Otherwise zaza won't run any tests.

Fixes: e19795ad8deafcddcd5fa08c7c1b400c1051e1af
Signed-off-by: Martin Kalcok <martin.kalcok@canonical.com>

mkalcok committed:
charm: Remove mantic support

Support for mantic (23.10) is removed as the release is pass
it's EOL.

Signed-off-by: Martin Kalcok <martin.kalcok@canonical.com>

fnordahl reviewed (approved):
LGTM, thanks!
